docs/source/core/core.rst

```diff
@@ -750,26 +750,34 @@ NeMo offers users two options for profiling: Nsys & CUDA memory profiling. These
 to debug performance issues as well as memory issues such as memory leaks.
 
 To enable Nsys profiling, add the following options to the model config:
-nsys_profile: False
-   start_step: 10  # Global batch to start profiling
-   end_step: 10 # Global batch to end profiling
-   ranks: [0] # Global rank IDs to profile
-   gen_shape: False # Generate model and kernel details including input shapes
 
-Finally, the model training script with:
+.. code-block:: yaml
+
+    nsys_profile: False
+        start_step: 10  # Global batch to start profiling
+        end_step: 10 # Global batch to end profiling
+        ranks: [0] # Global rank IDs to profile
+        gen_shape: False # Generate model and kernel details including input shapes
+
+Finally, run the model training script with:
+
+.. code-block:: bash
+
+    nsys profile -s none -o <profile filepath> -t cuda,nvtx --force-overwrite true --capture-range=cudaProfilerApi --capture-range-end=stop python ./examples/...
 
-nsys profile -s none -o <profile filepath> -t cuda,nvtx --force-overwrite true --capture-range=cudaProfilerApi --capture-range-end=stop python ./examples/... 
 See more options at `nsight user guide <https://docs.nvidia.com/nsight-systems/UserGuide/index.html#cli-profiling>`_.
 
 
 
 To enable CUDA memory profiling, add the following options to the model config:
 
-memory_profile:
-   enabled: True
-   start_step: 10  # Global batch to start profiling
-   end_step: 10 # Global batch to end profiling
-   rank: 0 # Global rank ID to profile
-   output_path: None # Path to store the profile output file
+.. code-block:: yaml
+
+    memory_profile:
+        enabled: True
+        start_step: 10  # Global batch to start profiling
+        end_step: 10 # Global batch to end profiling
+        rank: 0 # Global rank ID to profile
+        output_path: None # Path to store the profile output file
 
```

Comment:
I am providing line-by-line copyedits to this file here because most of the file is read only.

Line 7. fix capitalization.
NeMo models contain everything needed to train and reproduce conversational AI models:

Line 38. fix punctuation.
To see all available pretrained models for a specific NeMo model, use the ``list_available_models()`` method:

Line 55.fix spelling.
build complex AI applications without having to rewrite boilerplate code for PyTorch training.

Line 171. fix spelling.
PyTorch Lightning then handles all of the boilerplate code needed for training.

Lines 242-243. suggested revision.
The model configuration always contains ``train_ds``, ``validation_ds``, ``test_ds``, and ``optim``.  Model architectures, however, can vary across domains. Refer to the ASR, NLP, and TTS collections documentation for detailed information on model architecture configuration.

Line 293. fix formatting.
Delete line 292 and bring the :ref:`NeMo ASR Configuration Files` up so it appears after the colon (:)

Line 299. suggested revision.
of experiments on compute clusters or for quickly testing parameters during development.

Line 377. fix punctuation.
Configuration with Hydra always has the following precedence CLI > YAML > Dataclass.

Line 385. fix grammar.
configuration for a Novograd optimizer with a Cosine Annealing learning rate schedule.

Line 411. suggested revision. Remove the ..note:: and just leave: 
`NeMo Examples <https://github.com/NVIDIA/NeMo/tree/v1.0.2/examples>`_ has optimizer and scheduler configurations for every NeMo model.

Line 599. suggested revision.
Restoring conversational AI models can be complicated because it requires more than just the checkpoint weights; additional information is also needed to use the model effectively.

Line 646. fix capitalization.
Upload a Model to the Hub

Line 691 fix punctuation.
In some cases, it may be helpful to use NeMo models inside other NeMo models. For example, we can incorporate language models into ASR models to use in a decoding process to improve accuracy or use hybrid ASR-TTS models to generate audio from the text on the fly to train or fine-tune the ASR model.

Line 693. fix grammar.
There are three ways to instantiate child models inside parent models:

Line 699 fix run-on sentence.
To register a child model, use the register_nemo_submodule method of the parent model. This method will add the child model to a specified model attribute. During serialization, it will correctly handle child artifacts and store the child model’s configuration in the parent model’s config_field.

Line 749 fix grammar.
NeMo offers users two options for profiling: Nsys and CUDA memory profiling. These two options allow users
to debug performance issues as well as memory issues such as memory leaks.

Line 783. revise sentence. this is a suggestion. please review or update as needed.
Be sure to invoke your NeMo script without making any changes in the invocation command.

Reply:
> Line 293. fix formatting.
Delete line 292 and bring the :ref:NeMo ASR Configuration Files up so it appears after the colon (:)

I am just going to delete that text since technically we should point to all the other docs sections, not just ASR configuration files. But there's too many to point to.

Reply:
OK